lib/Dialect/TritonGPU/Transforms/Pipeliner/ScheduleLoops.cpp

```diff
@@ -190,8 +190,116 @@ bool hasLatenciesAssigned(scf::ForOp forOp,
   return false;
 }
 
+// Determine the chain of dots in the given set of users for a dot.
+std::tuple<SmallVector<ttng::MMAv5OpInterface>, bool>
+computeDotChain(ttng::MMAv5OpInterface dotOp,
```

Comment:
This is slightly modified from the first PR. I suggest we merge that first to simplify the review.